### ISSUE TITLE
Accept tcp protocol for DOCKER_URL

### ISSUE THREAD

**Issue by kinghuang** [joonas-fi/promswarmconnect#11]
Docker API endpoints are typically specified using `tcp` as the protocol rather than `http` or `https`. It would be nice if promswarmconnect accepted `tcp` in the `DOCKER_URL` value.

Right now, promswarmconnect throws an error if `tcp` is specified as the protocol.

```
Get tcp://docker-host-here:2376/v1.24/tasks?filters=%7B%22desired-state%22%3A%5B%22running%22%5D%7D: unsupported protocol scheme "tcp"
```

**Comment by joonas-fi** [comment]
You're right. It is weird the way Docker does it, since `http | https` is what goes on top of TCP (and TCP doesn't tell anything about the application-level protocol), but you're correct that we should use the same notation that Docker does it because it's confusing if promswarmconnect does it in a different way.

I need to investigate a bit more, but right now it seems that we'd need to behave this way:

- `tcp://` => http
- `tcp+tls://` => https

**Comment by kinghuang** [comment]
The Docker CLI and related tools always use `tcp://`. And then, the `DOCKER_TLS` environment variable is used to specify if TLS is on or not, and `DOCKER_TLS_VERIFY` for whether the cert is validated. They also use `DOCKER_HOST` to specify the Docker endpoint.

https://docs.docker.com/engine/reference/commandline/cli/#environment-variables

Ideally, I think `DOCKER_URL` should map tcp to http, as you've written. But, map tcp to https if `DOCKER_TLS` is set.
